src/plugins/intel_cpu/src/memory_desc/cpu_memory_desc_utils.cpp

```diff
@@ -45,40 +48,50 @@ DnnlBlockedMemoryDesc MemoryDescUtils::convertToDnnlBlockedMemoryDesc(const Memo
     }
 }
 
-CpuBlockedMemoryDesc MemoryDescUtils::convertToCpuBlockedMemoryDesc(const InferenceEngine::TensorDesc& desc) {
-    if (desc.getLayout() == InferenceEngine::Layout::ANY)
-        IE_THROW() << "Cannot convert InferenceEngine::TensorDesc with ANY layout to CpuBlockedMemoryDesc";
-
-    const auto& blkDesc = desc.getBlockingDesc();
-    const auto& dims = desc.getDims();
-
-    auto strides = blkDesc.getStrides();
-    // for empty tensor case InferenceEngine::TensorDesc fill strides with non zero values before first 0 dims
-    // i.e. dims[1, 0, 2, 3] -> strides [0, 6, 3, 1]
-    if (std::any_of(dims.begin(), dims.end(), [](size_t dim){ return dim == 0; })) {
-        std::fill(strides.begin(), strides.end(), 0);
-    }
-
-    return CpuBlockedMemoryDesc(desc.getPrecision(), Shape(dims), blkDesc.getBlockDims(), blkDesc.getOrder(), blkDesc.getOffsetPadding(),
-                                blkDesc.getOffsetPaddingToData(), strides);
+DnnlBlockedMemoryDesc MemoryDescUtils::createDnnlBlockedMemoryDesc(InferenceEngine::Precision prc,
+                                                                   const Shape& shape,
+                                                                   const VectorDims& blockedDims,
+                                                                   const VectorDims& blockedOrder) {
+    return DnnlBlockedMemoryDesc(prc, shape, blockedDims, blockedOrder);
 }
 
-DnnlBlockedMemoryDesc MemoryDescUtils::convertToDnnlBlockedMemoryDesc(const InferenceEngine::TensorDesc& desc) {
-    if (desc.getLayout() == InferenceEngine::Layout::ANY)
-        IE_THROW() << "Cannot convert InferenceEngine::TensorDesc with ANY layout to DnnlBlockedMemoryDesc";
-
-    const auto& blkDesc = desc.getBlockingDesc();
-    const auto& dims = desc.getDims();
-
-    auto strides = blkDesc.getStrides();
-    // for empty tensor case InferenceEngine::TensorDesc fill strides with non zero values before first 0 dims
-    // i.e. dims[1, 0, 2, 3] -> strides [0, 6, 3, 1]
-    if (std::any_of(dims.begin(), dims.end(), [](size_t dim){ return dim == 0; })) {
-        std::fill(strides.begin(), strides.end(), 0);
+CpuBlockedMemoryDesc MemoryDescUtils::createCpuBlockedMemoryDesc(const ov::SoPtr<ITensor>& tensor,
+                                                                 const bool canEmptyShape) {
```

Comment:
Most of time `canEmptyShape = true`, but  in `PushInputData` it cannot use empty shape input any more due to it will load input tensor data to internal memory, for such case it need set a proper shape according actual tensor buffer size.   `canEmptyShape' flag is used for whether to set a proper shape.

Reply:
As I understand it, it's the opposite. The plugin always works with scalars as first rank tensors. Thus any routine that converts the ngraph representation to the plugin specific representation have to respect this constraint.

Reply:
I encounter below failure to make me to do above changes:

```
[ RUN      ] smoke_MatMul_NoTranspose/MatMulLayerTest.Inference/IS=([]_[])_TS={(5)_(5)}_transpose_a=0_transpose_b=0_secondary_input_type=CONSTANT_modelType=f32_trgDev=CPUconfig=()

MEM_USAGE=472584KB
Access PluginCache ov core. OV Core use count: 0
Created ov core.
/mnt/data1_1c41/river/capi/openvino/src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp:90: Failure
Expected and actual shape are different: [] [1]
Reset PluginCache. OV Core use count: 1
[  FAILED  ] smoke_MatMul_NoTranspose/MatMulLayerTest.Inference/IS=([]_[])_TS={(5)_(5)}_transpose_a=0_transpose_b=0_secondary_input_type=CONSTANT_modelType=f32_trgDev=CPUconfig=(), where GetParam() = ({ ({}, { { 5 } }), ({}, { { 5 } }) }, (false, false), f32, CONSTANT, "CPU", {}) (147 ms)
[----------] 1 test from smoke_MatMul_NoTranspose/MatMulLayerTest (148 ms total)
```
Is there any idea to avoid above issue if we follow plugin's scalar policy?

Reply:
I have solved above issue, and remove `canEmptyShape` to follow plugin scalar data policy.